File: app/daa-sdk/java-sdk/src/main/java/com/daa/DaaClient.java
package com.daa;

import com.google.gson.Gson;
import java.io.PrintWriter;
import java.io.StringWriter;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Instant;
import java.util.HashMap;
import java.util.Map;

public class DaaClient {
    private final String backendUrl;
    private final String token;
    private final String appName;
    private final HttpClient httpClient;
    private final Gson gson;

    public DaaClient() {
        this(
            System.getenv("DAA_BACKEND_API_URL") != null ? System.getenv("DAA_BACKEND_API_URL") : "http://localhost:8000",
            System.getenv("DAA_TOKEN"),
            System.getenv("REPO_NAME") != null ? System.getenv("REPO_NAME") : "default-java-app"
        );
    }

    public DaaClient(String backendUrl, String token, String appName) {
        this.backendUrl = backendUrl;
        this.token = token;
        this.appName = appName;
        this.httpClient = HttpClient.newHttpClient();
        this.gson = new Gson();
    }

    public void captureException(Throwable throwable) {
        StringWriter sw = new StringWriter();
        PrintWriter pw = new PrintWriter(sw);
        throwable.printStackTrace(pw);
        String stackTrace = sw.toString();

        Map<String, Object> logContent = new HashMap<>();
        logContent.put("message", throwable.getMessage() != null ? throwable.getMessage() : throwable.toString());
        logContent.put("stack_trace", stackTrace);
        logContent.put("context", new HashMap<String, Object>());
        logContent.put("timestamp", Instant.now().toString());

        Map<String, String> payload = new HashMap<>();
        payload.put("content", gson.toJson(logContent));
        payload.put("app_name", this.appName);

        sendLog(payload);
    }

    public void sendLog(Map<String, String> payload) {
        try {
            String jsonPayload = gson.toJson(payload);
            HttpRequest.Builder requestBuilder = HttpRequest.newBuilder()
                    .uri(URI.create(this.backendUrl + "/logs/"))
                    .header("Content-Type", "application/json")
                    .POST(HttpRequest.BodyPublishers.ofString(jsonPayload));

            if (this.token != null && !this.token.isEmpty()) {
                requestBuilder.header("Authorization", "Bearer " + this.token);
            }

            HttpResponse<String> response = httpClient.send(requestBuilder.build(), HttpResponse.BodyHandlers.ofString());
            if (response.statusCode() < 200 || response.statusCode() >= 300) {
                System.err.println("DAA Java SDK received error status from backend: " + response.statusCode() + " - " + response.body());
            }
        } catch (Exception e) {
            System.err.println("DAA Java SDK failed to send log: " + e.getMessage());
        }
    }
}

File: app/daa-sdk/dotnet-sdk/DaaClient.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;

namespace Daa
{
    public class DaaClient
    {
        private readonly string _backendUrl;
        private readonly string _token;
        private readonly string _appName;
        private static readonly HttpClient HttpClient = new HttpClient();

        public DaaClient(string backendUrl = null, string token = null, string appName = null)
        {
            _backendUrl = backendUrl ?? Environment.GetEnvironmentVariable("DAA_BACKEND_API_URL") ?? "http://localhost:8000";
            _token = token ?? Environment.GetEnvironmentVariable("DAA_TOKEN");
            _appName = appName ?? Environment.GetEnvironmentVariable("REPO_NAME") ?? "default-dotnet-app";
        }

        public async Task CaptureExceptionAsync(Exception exception)
        {
            var logContent = new
            {
                message = exception.Message,
                stack_trace = exception.StackTrace ?? string.Empty,
                context = new Dictionary<string, object>(),
                timestamp = DateTime.UtcNow.ToString("o")
            };

            var payload = new
            {
                content = JsonSerializer.Serialize(logContent),
                app_name = _appName
            };

            await SendLogAsync(payload);
        }

        public async Task SendLogAsync(object payload)
        {
            try
            {
                var json = JsonSerializer.Serialize(payload);
                var content = new StringContent(json, Encoding.UTF8, "application/json");

                var request = new HttpRequestMessage(HttpMethod.Post, $"{_backendUrl}/logs/")
                {
                    Content = content
                };

                if (!string.IsNullOrEmpty(_token))
                {
                    request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", _token);
                }

                var response = await HttpClient.SendAsync(request);
                if (!response.IsSuccessStatusCode)
                {
                    var responseBody = await response.Content.ReadAsStringAsync();
                    Console.Error.WriteLine($"DAA .NET SDK error: {response.StatusCode} - {responseBody}");
                }
            }
            catch (Exception ex)
            {
                Console.Error.WriteLine($"DAA .NET SDK failed to send log: {ex.Message}");
            }
        }
    }
}
